kyuubi-server/src/main/scala/org/apache/kyuubi/server/metastore/MetadataStoreRequestsRetryRef.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.metastore
+
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.atomic.AtomicInteger
+
+class MetadataStoreRequestsRetryRef(val identifier: String) {
+  private[metastore] val retryCount = new AtomicInteger(0)
+
+  private[metastore] val retryingTaskCount = new AtomicInteger(0)
```

Comment:
seems one flag is enough, now `retryCount` only used for debug ?

Reply:
retryCount is only used for debug.

We should not set max retry count limitation.

Reply:
maybe we can set max retry time limitation.

Such as only retry it for `1 hour`.

Reply:
we can add a limitation for the queue, the max number of queue should be limited. otherwise it would cause oom